pkg/operator/ceph/csi/util.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+
+	apps "k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func loadTemplate(name, templatePath string) (string, error) {
+	b, err := ioutil.ReadFile(templatePath)
+	if err != nil {
+		return "", err
+	}
+	data := string(b)
+	var writer bytes.Buffer
+	t := template.New(name)
+	err = template.Must(t.Parse(data)).Execute(&writer, CSIParam)
+	return writer.String(), err
+}
+
+func templateToStatefulSet(name, templatePath string) (*apps.StatefulSet, error) {
+	var ss apps.StatefulSet
+	t, err := loadTemplate(name, templatePath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = yaml.Unmarshal([]byte(t), &ss)
+	if err != nil {
+		return nil, err
+	}
+	return &ss, nil
+}
+
+func templateToDaemonSet(name, templatePath string) (*apps.DaemonSet, error) {
+	var ds apps.DaemonSet
+	t, err := loadTemplate(name, templatePath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = yaml.Unmarshal([]byte(t), &ds)
+	if err != nil {
+		return nil, err
+	}
+	return &ds, nil
+}
+
+func makeHeadlessSvc(name, namespace string, label map[string]string, clientset kubernetes.Interface) (*corev1.Service, error) {
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    label,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: label,
+			Ports:    []corev1.ServicePort{{Name: "dummy", Port: dummyPort}},
+		},
+	}
+
+	_, err := clientset.CoreV1().Services(namespace).Create(svc)
```

Comment:
Some of these helpers might exist in the package `k8sutil`, and you could add these helpers to that package if they're not already there.

Reply:
done